subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/gcc/GccPlatformToolProvider.java

```diff
@@ -138,6 +152,11 @@
         return new ArStaticLibraryArchiver(buildOperationExecutor, commandLineTool(staticLibArchiverTool), context(staticLibArchiverTool), workerLeaseService);
     }
 
+    private GccVersionResult gccMetadata(GccCommandLineToolConfigurationInternal cppCompilerTool) {
+        File gccLocation = toolSearchPath.locate(cppCompilerTool.getToolType(), cppCompilerTool.getExecutable()).getTool();
+        return metaDataProvider.getGccMetaData(gccLocation, ((DefaultGccPlatformToolChain) toolRegistry).getCompilerProbeArgs());
```

Comment:
Should we cache this metadata somehow?

Reply:
We already do, since `metaDataProvider` is a `CachingCompilerMetaDataProvider`.